int16 s1_dist
int16 s2_dist
int16 s3_dist
int16 s4_dist
int16 s5_dist
